include/libtorrent/sha1_hash.hpp

```diff
@@ -262,7 +263,10 @@ namespace libtorrent
 		{
 			return std::string(reinterpret_cast<char const*>(&m_number[0]), size());
 		}
-
+		std::string to_hex() const
```

Comment:
I would expect it to work better if that function would just take a ``span<char const>``. the ``span`` has (properly constrained) template constructor from anything that looks like a container.